Modules/ZDC/src/ZDCRecDataCheck.cxx

```diff
@@ -0,0 +1,321 @@
+          mNumWADC += 1;
+          mStringWADC = mStringWADC + mVectParamADC.at(i).ch + " ";
+          ILOG(Warning, Support) << "Rec Warning in " << mVectParamADC.at(i).ch << " intervall: " << mVectParamADC.at(i).minW << " - " << mVectParamADC.at(i).maxW << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+        if (((float)h->GetBinContent(ib) < (float)mVectParamADC.at(i).minE) || ((float)h->GetBinContent(ib) > (float)mVectParamADC.at(i).maxE)) {
+          mNumEADC += 1;
+          mStringEADC = mStringEADC + mVectParamADC.at(i).ch + " ";
+          ILOG(Error, Support) << "Rec Error in " << mVectParamADC.at(i).ch << " intervall: " << mVectParamADC.at(i).minE << " - " << mVectParamADC.at(i).maxE << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+      }
+      if (mNumWADC == 0 && mNumEADC == 0) {
+        mQADC = 1;
+      }
+      if (mNumWADC > 0) {
+        mQADC = 2;
+      }
+      if (mNumEADC > 0) {
+        mQADC = 3;
+      }
+    }
+    mNumETDC = 0;
+    mNumWTDC = 0;
+    mStringWTDC = "";
+    mStringETDC = "";
+    if (mo->getName() == "h_summmary_TDC") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+      // dumpVecParam((int)h->GetNbinsX(),(int)mVectParamTDC.size());
+      if ((int)h->GetNbinsX() != (int)mVectParamTDC.size())
+        return Quality::Null;
+      for (int i = 0; i < h->GetNbinsX(); i++) {
+        ib = i + 1;
+        if ((((float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).minW && (float)h->GetBinContent(ib) >= (float)mVectParamTDC.at(i).minE)) || ((float)h->GetBinContent(ib) > (float)mVectParamTDC.at(i).maxW && (float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).maxE)) {
+          mNumWTDC += 1;
+          mStringWTDC = mStringWTDC + mVectParamTDC.at(i).ch + " ";
+          ILOG(Warning, Support) << "Rec Warning in " << mVectParamTDC.at(i).ch << " intervall: " << mVectParamTDC.at(i).minW << " - " << mVectParamTDC.at(i).maxW << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+        if (((float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).minE) || ((float)h->GetBinContent(ib) > (float)mVectParamTDC.at(i).maxE)) {
+          mNumETDC += 1;
+          mStringETDC = mStringETDC + mVectParamTDC.at(i).ch + " ";
+          ILOG(Error, Support) << "Rec Error in " << mVectParamTDC.at(i).ch << " intervall: " << mVectParamTDC.at(i).minE << " - " << mVectParamTDC.at(i).maxE << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+      }
+      if (mNumWTDC == 0 && mNumETDC == 0) {
+        mQTDC = 1;
+      }
+      if (mNumWTDC > 0) {
+        mQTDC = 2;
+      }
+      if (mNumETDC > 0) {
+        mQTDC = 3;
+      }
+
+      if (mQADC == 1 && mQTDC == 1) {
+        result = Quality::Good;
+        result.addReason(FlagReasonFactory::Unknown(), "ZDCRecDataTask quality is Good");
```

Comment:
as said before, I would remove it, since the flag "Unknown" is to mark data as bad, but you cannot say why it is bad.